lib/jsdom/browser/Window.js

```diff
@@ -642,6 +642,20 @@ function Window(options) {
 
   this.getComputedStyle = function (elt) {
     elt = Element.convert(elt);
+    let pseudoElt = arguments[1];
+    if (pseudoElt !== undefined && pseudoElt !== null) {
+      pseudoElt = webIDLConversions.DOMString(pseudoElt);
+    }
+
+    if (pseudoElt !== undefined && pseudoElt !== null && pseudoElt !== "") {
+      // TODO: Parse pseudoElt
+
+      notImplemented("window.computedStyle(elt, pseudoElt)", this);
+
+      if (SHADOW_DOM_PSEUDO_REGEXP.test(pseudoElt)) {
+        throw new TypeError("Tried to get the computed style of a Shadow DOM pseudo-element.");
+      }
```

Comment:
I don't really understand this. Doesn't `notImplemented` already throw an error, and thus make this if-statement dead code?

Reply:
No, `notImplemented` simply logs a warning to the console: <https://github.com/jsdom/jsdom/blob/7242c4b238694a75b1a975572b767626dd550873/lib/jsdom/browser/not-implemented.js#L3-L13>